File: types/crm.ts
export type StatusTone = "neutral" | "good" | "warn" | "danger";

export type KpiCard = {
  label: string;
  value: string;
  change: string;
  tone: StatusTone;
};

export type ActivityItem = {
  title: string;
  subtitle: string;
  time: string;
  tone: StatusTone;
};

export type StageCard = {
  stage: string;
  count: number;
  value: string;
  nextAction: string;
};

export type LeadItem = {
  id: string;
  childName: string;
  parentName: string;
  source: string;
  stage: string;
  classInterest: string;
  followUp: string;
};

export type LeadDetail = {
  id: string;
  childName: string;
  parentName: string;
  parentEmail?: string;
  parentPhone?: string;
  source: string;
  stage: string;
  classInterest: string;
  followUp: string;
  notes: string;
  createdAt: string;
};

export type FamilyDetail = {
  id: string;
  householdName: string;
  guardians: string[];
  students: string[];
  balance: string;
  status: string;
  lastPayment: string;
  openTickets: number;
  notes: string;
};

export type StudentDetail = {
  id: string;
  fullName: string;
  className: string;
  guardian: string;
  attendance: string;
  behaviour: string;
  medicalFlag: string;
  nextAssessment: string;
  feeStatus: string;
  documents: string[];
};

export type ParentItem = {
  id: string;
  name: string;
  relationship: string;
  studentName: string;
  phone: string;
  email: string;
  status: string;
};

export type ParentDetail = {
  id: string;
  name: string;
  relationship: string;
  phone: string;
  email: string;
  preferredChannel: string;
  students: string[];
  lastActivity: string;
  notes: string;
};

export type HouseholdItem = {
  id: string;
  householdName: string;
  guardians: string[];
  students: number;
  balance: string;
  status: string;
};

export type StudentItem = {
  id: string;
  fullName: string;
  className: string;
  guardian: string;
  attendance: string;
  behaviour: string;
  medicalFlag: string;
};

export type FinanceSummary = {
  totalBilled: string;
  totalCollected: string;
  overdue: string;
  collectionRate: string;
};

export type InvoiceItem = {
  id: string;
  student: string;
  term: string;
  amountDue: string;
  amountPaid: string;
  dueDate: string;
  status: string;
};

export type DebtorItem = {
  student: string;
  className: string;
  balance: string;
  aging: string;
  lastContact: string;
};

export type MessagingMetric = {
  channel: string;
  sent: string;
  openRate: string;
  delivery: string;
};

export type CampaignItem = {
  title: string;
  audience: string;
  channel: string;
  status: string;
  sentAt: string;
};

export type TicketItem = {
  id: string;
  subject: string;
  parent: string;
  priority: string;
  assignee: string;
  sla: string;
  status: string;
};

export type TicketDetail = {
  id: string;
  subject: string;
  parent: string;
  priority: string;
  assignee: string;
  sla: string;
  status: string;
  description: string;
  createdAt: string;
  timeline: Array<{ time: string; note: string }>;
};

export type InvoiceDetailLineItem = {
  code: string;
  description: string;
  amount: string;
};

export type PaymentRecord = {
  method: string;
  amount: string;
  paidAt: string;
};

export type InvoiceDetail = {
  id: string;
  student: string;
  term: string;
  amountDue: string;
  amountPaid: string;
  dueDate: string;
  status: string;
  issuedAt: string;
  lineItems: InvoiceDetailLineItem[];
  payments: PaymentRecord[];
};

export type FeeStructureItem = {
  id: string;
  className: string;
  termName: string;
  title: string;
  amount: string;
  dueDays: number;
};

export type MessageTemplateItem = {
  id: string;
  name: string;
  channel: string;
  useCase: string;
  lastEdited: string;
};

export type StaffMetric = {
  label: string;
  value: string;
  note: string;
};

export type StaffMember = {
  name: string;
  role: string;
  attendance: string;
  responseTime: string;
  performance: string;
};

export type ReportPoint = {
  name: string;
  value: number;
};

export type ReportCard = {
  title: string;
  insight: string;
  value: string;
};

export type SettingGroup = {
  title: string;
  description: string;
  items: Array<{
    label: string;
    value: string;
  }>;
};

export type AuthPayload = {
  email: string;
  password: string;
};

export type ForgotPasswordPayload = {
  email: string;
};

export type ResetPasswordPayload = {
  token: string;
  newPassword: string;
};

export type ConvertLeadPayload = {
  familyName: string;
  primaryGuardian: string;
  studentFirstName: string;
  studentLastName: string;
  studentGender?: string;
  dateOfBirth?: string;
  admittedClass?: string;
};

export type LeadCreateRequest = {
  firstName: string;
  lastName: string;
  parentName: string;
  parentPhone: string;
  parentEmail?: string;
  source: string;
  stage: string;
  interestedClass?: string;
  followUpAt?: string;
};

export type PasswordResetResponse = {
  message: string;
};

export type AuthResponse = {
  access_token: string;
  refresh_token: string;
  token_type: "bearer";
  expires_in: number;
  user: {
    id: string;
    schoolId: string;
    role: string;
    fullName: string;
    email: string;
  };
};

export type DashboardResponse = {
  schoolName: string;
  sessionLabel: string;
  kpis: KpiCard[];
  pipeline: StageCard[];
  activity: ActivityItem[];
  revenueTrend: ReportPoint[];
};

export type AdmissionsResponse = {
  leads: LeadItem[];
  pipeline: StageCard[];
};

export type FamiliesResponse = {
  households: HouseholdItem[];
};

export type StudentsResponse = {
  students: StudentItem[];
};

export type ParentsResponse = {
  parents: ParentItem[];
};

export type TicketDetailResponse = {
  id: string;
  subject: string;
  parent: string;
  priority: string;
  assignee: string;
  sla: string;
  status: string;
  description: string;
  createdAt: string;
  timeline: Array<{ time: string; note: string }>;
};

export type InvoiceDetailResponse = {
  id: string;
  student: string;
  term: string;
  amountDue: string;
  amountPaid: string;
  dueDate: string;
  status: string;
  issuedAt: string;
  lineItems: InvoiceDetailLineItem[];
  payments: PaymentRecord[];
};

export type FeeStructuresResponse = {
  items: FeeStructureItem[];
};

export type MessageTemplatesResponse = {
  templates: MessageTemplateItem[];
};

export type ConvertLeadResponse = {
  leadId: string;
  familyId: string;
  studentId: string;
  status: string;
};

export type FinanceResponse = {
  summary: FinanceSummary;
  invoices: InvoiceItem[];
  debtors: DebtorItem[];
};

export type MessagingResponse = {
  metrics: MessagingMetric[];
  campaigns: CampaignItem[];
};

export type HelpdeskResponse = {
  tickets: TicketItem[];
};

export type StaffResponse = {
  metrics: StaffMetric[];
  people: StaffMember[];
};

export type ReportsResponse = {
  cards: ReportCard[];
  admissionsTrend: ReportPoint[];
  collectionsTrend: ReportPoint[];
};

export type SettingsResponse = {
  groups: SettingGroup[];
};

File: app/schemas/crm.py
from pydantic import BaseModel, EmailStr


class AuthRequest(BaseModel):
    email: EmailStr
    password: str


class AuthRefreshRequest(BaseModel):
    refresh_token: str


class ForgotPasswordRequest(BaseModel):
    email: EmailStr


class ResetPasswordRequest(BaseModel):
    token: str
    newPassword: str


class AuthUser(BaseModel):
    id: str
    schoolId: str
    role: str
    fullName: str
    email: EmailStr


class AuthResponse(BaseModel):
    access_token: str
    refresh_token: str
    token_type: str = "bearer"
    expires_in: int
    user: AuthUser


class KpiCard(BaseModel):
    label: str
    value: str
    change: str
    tone: str


class ActivityItem(BaseModel):
    title: str
    subtitle: str
    time: str
    tone: str


class StageCard(BaseModel):
    stage: str
    count: int
    value: str
    nextAction: str


class DashboardResponse(BaseModel):
    schoolName: str
    sessionLabel: str
    kpis: list[KpiCard]
    pipeline: list[StageCard]
    activity: list[ActivityItem]
    revenueTrend: list[dict[str, float | str]]


class LeadItem(BaseModel):
    id: str
    childName: str
    parentName: str
    source: str
    stage: str
    classInterest: str
    followUp: str


class LeadDetailResponse(BaseModel):
    id: str
    childName: str
    parentName: str
    parentEmail: str | None = None
    parentPhone: str | None = None
    source: str
    stage: str
    classInterest: str
    followUp: str
    notes: str
    createdAt: str


class ConvertLeadRequest(BaseModel):
    familyName: str
    primaryGuardian: str
    studentFirstName: str
    studentLastName: str
    studentGender: str | None = None
    dateOfBirth: str | None = None
    admittedClass: str | None = None


class ConvertLeadResponse(BaseModel):
    leadId: str
    familyId: str
    studentId: str
    status: str


class LeadCreateRequest(BaseModel):
    firstName: str
    lastName: str
    parentName: str | None = None
    parentPhone: str | None = None
    parentEmail: EmailStr | None = None
    source: str
    stage: str
    interestedClass: str | None = None
    followUpAt: str | None = None


class LeadUpdateRequest(BaseModel):
    stage: str | None = None
    followUpAt: str | None = None
    lostReason: str | None = None


class InvoiceLineItemRequest(BaseModel):
    code: str
    description: str
    amount: float


class InvoiceCreateRequest(BaseModel):
    studentId: str
    term: str
    amountDue: float
    dueDate: str
    lineItems: list[InvoiceLineItemRequest]


class PaymentRecord(BaseModel):
    invoiceId: str
    amount: float
    method: str
    reference: str | None = None
    paidAt: str


class PaystackInitResponse(BaseModel):
    authorization_url: str
    reference: str


class FlutterwaveInitResponse(BaseModel):
    checkout_url: str
    reference: str


class BroadcastRequest(BaseModel):
    audience: str
    channel: str
    templateId: str
    message: str | None = None


class TicketCreateRequest(BaseModel):
    parentId: str | None = None
    familyId: str | None = None
    subject: str
    description: str | None = None
    priority: str
    assigneeId: str | None = None
    slaDueAt: str | None = None


class TicketUpdateRequest(BaseModel):
    status: str | None = None
    priority: str | None = None
    assigneeId: str | None = None
    slaDueAt: str | None = None


class ReportResponse(BaseModel):
    reportName: str
    data: list[dict[str, str | int | float]]


class AdmissionsResponse(BaseModel):
    leads: list[LeadItem]
    pipeline: list[StageCard]


class HouseholdItem(BaseModel):
    id: str
    householdName: str
    guardians: list[str]
    students: int
    balance: str
    status: str


class FamilyDetailResponse(BaseModel):
    id: str
    householdName: str
    guardians: list[str]
    students: list[str]
    balance: str
    status: str
    lastPayment: str
    openTickets: int
    notes: str


class ParentItem(BaseModel):
    id: str
    name: str
    relationship: str
    studentName: str
    phone: str
    email: str
    status: str


class ParentDetailResponse(BaseModel):
    id: str
    name: str
    relationship: str
    phone: str
    email: EmailStr
    preferredChannel: str
    students: list[str]
    lastActivity: str
    notes: str


class FamiliesResponse(BaseModel):
    households: list[HouseholdItem]


class StudentItem(BaseModel):
    id: str
    fullName: str
    className: str
    guardian: str
    attendance: str
    behaviour: str
    medicalFlag: str


class StudentDetailResponse(BaseModel):
    id: str
    fullName: str
    className: str
    guardian: str
    attendance: str
    behaviour: str
    medicalFlag: str
    nextAssessment: str
    feeStatus: str
    documents: list[str]


class StudentsResponse(BaseModel):
    students: list[StudentItem]


class ParentsResponse(BaseModel):
    parents: list[ParentItem]


class FinanceSummary(BaseModel):
    totalBilled: str
    totalCollected: str
    overdue: str
    collectionRate: str


class InvoiceItem(BaseModel):
    id: str
    student: str
    term: str
    amountDue: str
    amountPaid: str
    dueDate: str
    status: str


class DebtorItem(BaseModel):
    student: str
    className: str
    balance: str
    aging: str
    lastContact: str


class FinanceResponse(BaseModel):
    summary: FinanceSummary
    invoices: list[InvoiceItem]
    debtors: list[DebtorItem]


class PaymentItem(BaseModel):
    id: str
    invoiceId: str
    student: str
    amount: str
    method: str
    paidAt: str
    status: str


class PaymentsResponse(BaseModel):
    payments: list[PaymentItem]


class InvoiceListResponse(BaseModel):
    invoices: list[InvoiceItem]


class BroadcastItem(BaseModel):
    id: str
    title: str
    audience: str
    channel: str
    status: str
    scheduledAt: str


class BroadcastsResponse(BaseModel):
    broadcasts: list[BroadcastItem]


class CalendarEvent(BaseModel):
    id: str
    title: str
    date: str
    time: str
    location: str
    status: str


class CalendarResponse(BaseModel):
    events: list[CalendarEvent]


class FeeStructureItem(BaseModel):
    id: str
    className: str
    termName: str
    title: str
    amount: str
    dueDays: int


class FeeStructuresResponse(BaseModel):
    items: list[FeeStructureItem]


class MessageTemplateItem(BaseModel):
    id: str
    name: str
    channel: str
    useCase: str
    lastEdited: str


class MessageTemplatesResponse(BaseModel):
    templates: list[MessageTemplateItem]


class MessagingMetric(BaseModel):
    channel: str
    sent: str
    openRate: str
    delivery: str


class CampaignItem(BaseModel):
    title: str
    audience: str
    channel: str
    status: str
    sentAt: str


class MessagingResponse(BaseModel):
    metrics: list[MessagingMetric]
    campaigns: list[CampaignItem]


class TicketItem(BaseModel):
    id: str
    subject: str
    parent: str
    priority: str
    assignee: str
    sla: str
    status: str


class TicketDetailResponse(BaseModel):
    id: str
    subject: str
    parent: str
    priority: str
    assignee: str
    sla: str
    status: str
    description: str
    createdAt: str
    timeline: list[dict[str, str]]


class HelpdeskResponse(BaseModel):
    tickets: list[TicketItem]


class InvoiceDetailLineItem(BaseModel):
    code: str
    description: str
    amount: str


class PaymentRecord(BaseModel):
    method: str
    amount: str
    paidAt: str


class InvoiceDetailResponse(BaseModel):
    id: str
    student: str
    term: str
    amountDue: str
    amountPaid: str
    dueDate: str
    status: str
    issuedAt: str
    lineItems: list[InvoiceDetailLineItem]
    payments: list[PaymentRecord]


class StaffMetric(BaseModel):
    label: str
    value: str
    note: str


class StaffMember(BaseModel):
    name: str
    role: str
    attendance: str
    responseTime: str
    performance: str


class StaffResponse(BaseModel):
    metrics: list[StaffMetric]
    people: list[StaffMember]


class ReportCard(BaseModel):
    title: str
    insight: str
    value: str


class ReportsResponse(BaseModel):
    cards: list[ReportCard]
    admissionsTrend: list[dict[str, float | str]]
    collectionsTrend: list[dict[str, float | str]]


class SettingItem(BaseModel):
    label: str
    value: str


class SettingGroup(BaseModel):
    title: str
    description: str
    items: list[SettingItem]


class SettingsResponse(BaseModel):
    groups: list[SettingGroup]
